### PR TITLE
doc/cephfs: repairing inaccessible FSes

### PR DESCRIPTION
**Pull request by zdover23**
Add a procedure to doc/cephfs/troubleshooting.rst that explains how to restore access to FileSystems that became inaccessible after post-Nautilus upgrades. The procedure included here was written by Harry G Coin, and merely lightly edited by me. I include him here as a "co-author", but it should be noted that he did the heavy lifting on this.

See the email thread here for more context:
https://lists.ceph.io/hyperkitty/list/ceph-users@ceph.io/message/HS5FD3QFR77NAKJ43M2T5ZC25UYXFLNW/

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/cephfs/troubleshooting.rst [hunk 1]
```diff
@@ -222,6 +222,58 @@ The In-memory Log Dump can be disabled using::
 
   $ ceph config set mds mds_extraordinary_events_dump_interval 0
 
+Filesystems Become Inaccessible After an Upgrade
+================================================
+
+IF
+
+you have Ceph file systems that have data and metadata pools that were
```
Comment (anthonyeleven) [inline doc/cephfs/troubleshooting.rst:230]: Suggest s/Ceph/CephFS/ to disambiguate

File: doc/cephfs/troubleshooting.rst [hunk 2]
```diff
@@ -222,6 +222,58 @@ The In-memory Log Dump can be disabled using::
...
+specified in the ``ceph fs new`` command (meaning that they were not created
```
Comment (anthonyeleven) [inline doc/cephfs/troubleshooting.rst:231]: created by a

File: doc/cephfs/troubleshooting.rst [hunk 3]
```diff
@@ -222,6 +222,58 @@ The In-memory Log Dump can be disabled using::
...
+with the defaults)
+
+OR
+
+you have an existing ceph file system and are upgrading to a new post-Nautilus
```
Comment (anthonyeleven) [inline doc/cephfs/troubleshooting.rst:236]: CephFS file system

**Review by anthonyeleven — APPROVED** [review]
A few suggestions inline, plus a note that one might prima fascie think it shouldn't be backported, but I suggest that doing so will help people avoid the situation in the first place.

**Comment by anthonyeleven** [comment]
Might we also want to add words to the effect that performing this step *before* such an upgrade is advisable to prevent the problem in the first place?

**Comment by zdover23** [comment]
> A few suggestions inline, plus a note that one might prima fascie think it shouldn't be backported, but I suggest that doing so will help people avoid the situation in the first place.

I'm considering backporting this to all of the branches from Nautilus to Reef inclusive.

File: doc/cephfs/troubleshooting.rst (hunk 1 above)
Comment (zdover23) [reply doc/cephfs/troubleshooting.rst:230]: Agreed. Changed.

File: doc/cephfs/troubleshooting.rst (hunk 2 above)
Comment (zdover23) [reply doc/cephfs/troubleshooting.rst:231]: Thanks for the fresh-eyed read. I've adopted this. It's better.

File: doc/cephfs/troubleshooting.rst (hunk 3 above)
Comment (zdover23) [reply doc/cephfs/troubleshooting.rst:236]: Good catch.

**Comment by zdover23** [comment]
> Might we also want to add words to the effect that performing this step _before_ such an upgrade is advisable to prevent the problem in the first place?

I think that the existence of this PR shows that the whole CephFS documentation suite post-Nautilus should be reviewed with the goal of at the very least using documentation to fix this compatibility-breaking update. However, it'll be a few months before we get to the CephFS docs review given our current manpower. So, in short: yeah. I'll add a note.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/51371 - Reef backport
https://github.com/ceph/ceph/pull/51372 - Quincy backport
https://github.com/ceph/ceph/pull/51373 - Pacific backport

**Pull request merged**